instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/internal/BedrockRuntimeImpl.java

```diff
@@ -1362,17 +1514,71 @@ private void handleEventAnthropicClaude(Document result) {
             }
             return;
           }
+        case "content_block_start":
+          {
+            Document contentBlock = result.asMap().get("content_block");
+            if (contentBlock == null || !contentBlock.isMap()) {
+              return;
+            }
+            Document contentBlockType = contentBlock.asMap().get("type");
+            if (contentBlockType == null || !contentBlockType.isString()) {
+              return;
+            }
+            if (contentBlockType.asString().equals("tool_use")) {
+              currentTool = ToolUseBlock.builder();
+              handleToolUseAnthropicCloud(contentBlock, currentTool);
+            }
+            return;
+          }
         case "content_block_delta":
           {
             Document delta = result.asMap().get("delta");
             if (delta == null || !delta.isMap()) {
               return;
             }
-            if (captureMessageContent) {
-              Document text = delta.asMap().get("text");
-              if (text != null && text.isString()) {
-                currentText.append(text.asString());
+            Document deltaType = delta.asMap().get("type");
+            if (deltaType == null || !deltaType.isString()) {
+              return;
+            }
+            switch (deltaType.asString()) {
+              case "text_delta":
+                {
+                  if (captureMessageContent) {
+                    Document text = delta.asMap().get("text");
+                    if (text != null && text.isString()) {
+                      currentText.append(text.asString());
+                    }
+                  }
+                  return;
+                }
+              case "input_json_delta":
+                {
+                  if (currentInputJson == null) {
+                    currentInputJson = new StringBuilder();
+                  }
```

Comment:
if `partial_json` does not exist then in `content_block_stop` `currentInputJson` would be an empty string that appears to trigger a NPE in `deserializeDocument` because `node` is `null`

Reply:
Thanks! Moved the initialization to be more lazy.